### PR TITLE
Enumerate Columns According to Horizontal Model Dimensions

### PR DESCRIPTION
**Pull request by bska**
As mentioned in #2331, the restart file's enumeration of active cells connected to analytic aquifers follows a column-based scheme.  Recent work on compatibility prompted by real field models with the constant flux analytic aquifer type (see also PRs #3378 and OPM/opm-simulators#4429) has shown that the enumeration of the columns also must account for the horizontal model dimension (keyword `DIMENS`).  If `Nx >= Ny` (`DIMENS(1) >= DIMENS(2)`) then the columns are enumerated with the `J` index cycling faster than the `I` index.  On the other hand, when `Ny > Nx` the `I` index is supposed to cycle more quickly than the `J` index.

This PR guarantees that he innermost loop is always across the model layers while we ensure that the 'outer' loop always iterates over an index range that is at least as large as that of the 'middle' loop.

Thanks a lot to [at]tskille for the new insight.

### CODE REVIEW

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
The [regression failures]() are expected because in both cases (`AQUFLUX-01`, [`DIMENS = 8 15 3`](https://github.com/OPM/opm-tests/blob/c082e0904ebdbbf8cadcd56fb8331064bcba14d1/aquifers/AQUFLUX-01.DATA#L22-L23) and `0A_AQUCT_MODEL6` [`DIMENS = 12 14 15`](https://github.com/OPM/opm-tests/blob/c082e0904ebdbbf8cadcd56fb8331064bcba14d1/model6/0A_AQUCT_MODEL6.DATA#L43-L44)) we have `Ny > Nx`.  That's the exact situation that's affected here.  We'll need to update those reference solutions once this PR is approved.

Nevertheless, I'm setting the PR back to draft state to enable additional testing before approval.

**Comment by bska** [comment]
This has been verified to produce the expected outputs on a real field case.  I'm marking it as "ready for review".  I'll run a build check as well which will fail for the two regression tests

* [`AQUFLUX-01`](https://github.com/OPM/opm-tests/blob/325752d8dfed47e0a0aa2ec963a85e359cfa9936/aquifers/AQUFLUX-01.DATA#L19-L23)
* [`0A_AQUCT_MODEL6`](https://github.com/OPM/opm-tests/blob/325752d8dfed47e0a0aa2ec963a85e359cfa9936/model6/0A_AQUCT_MODEL6.DATA#L43-L44)

We must update the reference solutions for these cases as the contents of the `ICAQ` restart file array will change.  Any other failures are unexpected and must be resolved before the PR is eligible for merging.

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
>  I'll run a build check as well which will fail for the two regression tests

Those two regression tests [did fail](https://ci.opm-project.org/job/opm-common-PR-builder/5567/testReport/), and for the expected reason&mdash;solution vectors unchanged, but `ICAQ` contents altered.  No other regression tests failed here so this PR is safe to merge once approved and we've generated new reference solutions for the two test cases.

**Review by blattms — APPROVED** [review]
Looks good to me.

**Comment by bska** [comment]
> Looks good to me.

Thank you for taking the time to review the PR, @blattms.  Based on that assessment I will update the reference solutions of the tow regression cases in question and merge this once the final build check is green.

**Comment by bska** [comment]
jenkins build this update_data please

**Comment by bska** [comment]
jenkins build this opm-tests=917 please

**Pull request merged**
